src/extension.ts

```diff
@@ -2866,7 +2878,7 @@ const buildFlashAndMonitor = async (runMonitor: boolean = true) => {
 async function selectFlashMethod(cancelToken) {
   let flashType = idfConf.readParameter("idf.flashType");
   if (!flashType) {
-    flashType = await vscode.window.showQuickPick(["JTAG", "UART"], {
+    flashType = await vscode.window.showQuickPick(["JTAG", "UART", "DFU"], {
```

Comment:
Can reuse the enum types as string array to give the options to `showQuickPick` method.